lib/Dialect/SV/Ops.cpp

```diff
@@ -113,6 +114,54 @@ static void printModportStructs(OpAsmPrinter &p, Operation *,
   p << ')';
 }
 
+//===----------------------------------------------------------------------===//
+// Extract operation.
+//===----------------------------------------------------------------------===//
+
+static ParseResult parseExtractOp(OpAsmParser &p, OperationState &result) {
+  MLIRContext *ctxt = result.getContext();
+  FlatSymbolRefAttr field;
+  OpAsmParser::OperandType extractable;
+  Type inputType;
+  if (p.parseAttribute(field) || p.parseOperand(extractable) || p.parseColon())
+    return failure();
+  result.addAttribute("field", field);
+
+  auto typeLoc = p.getCurrentLocation();
+  if (p.parseType(inputType))
+    return failure();
+
+  if (p.resolveOperand(extractable, inputType, result.operands))
+    return failure();
+
+  // The type specified is the extractable type, so the return type can be
+  // inferred based on that. Said resolution is different depending on the
+  // extractable type.
+  if (auto ifaceTy = inputType.dyn_cast<InterfaceType>()) {
+    SymbolRefAttr ifaceSym = ifaceTy.getInterface();
```

Comment:
This is the only way I could figure out to append a symbol to a pre-existing symbol ref. Surely there's a better way?

Reply:
Correct me if I'm wrong, but won't the `ifaceSym` only have a flat reference to the interface's symbolic name?

If so, couldn't this boil down to the following?

```
SymbolRefAttr::get(ifaceSym.getValue(), {field}, ctx);
```

Reply:
I don't know if it can be a flat reference. The InterfaceOp may be defined in a different symbol table. It is simpler to require/assume that everything is defined in a single symbol table to model the SystemVerilog global namespace. SystemVerilog packages could be modeled as a sub symbol table, but we're not there yet.

Reply:
Ended up going with a flat reference. We can cross the symbol table bridge when we come to it.